multi_cgra/test/MeshMultiCgraRTL_test.py

```diff
@@ -307,18 +313,25 @@ def test_sim_homo_2x2_2x2(cmdline_opts):
   th = config_model_with_cmdline_opts(th, cmdline_opts, duts = ['dut'])
   run_sim(th)
 
-# def test_verilog_homo_2x2_4x4(cmdline_opts):
-#   th = initialize_test_harness(cmdline_opts,
-#                                num_cgra_rows = 2,
-#                                num_cgra_columns = 2,
-#                                num_x_tiles_per_cgra = 4,
-#                                num_y_tiles_per_cgra = 4,
-#                                num_banks_per_cgra = 8,
-#                                data_mem_size_per_bank = 256)
-#   th.elaborate()
-#   th.dut.set_metadata(VerilogVerilatorImportPass.vl_Wno_list,
-#                       ['UNSIGNED', 'UNOPTFLAT', 'WIDTH', 'WIDTHCONCAT',
-#                        'ALWCOMBORDER'])
-#   th = config_model_with_cmdline_opts(th, cmdline_opts, duts = ['dut'])
-#   th.apply(DefaultPassGroup())
+def _enable_translate_recursively(m):
+  m.set_metadata(VerilogTranslationPass.enable, True)
+  for child in m.get_child_components(repr):
+    _enable_translate_recursively( child )
+
+def translate_model(top, submodules_to_translate):
```

Comment:
The main purpose here is to skip the `apply(DefaultPassGroup()`, right? And `top.apply(VerilogPlaceholderPass())` is necessary for this purpose? Just curious why we need a placeholderPass here..

Reply:
The main purpose is to apply `VerilogTranslationPass` individually, instead of `VerilogTranslationImportPass`.

`VerilogPlaceholderPass` is for embedded Verilog modules, i.e. if you have imported modules in your design `VerilogPlaceholderPass` will pickle its source file I think.

Reply:
Plz change
https://github.com/tancheng/VectorCGRA/blob/c08b879c3942ffaf281b42080b653f7563b3138e/.github/workflows/python-package.yml#L84
to
```
pytest ../multi_cgra/test/MeshMultiCgraRTL_test.py::test_verilog_homo_2x2_4x4 -xvs --test-verilog --dump-vtb --dump-vcd
```
to enable the feature provided in this PR. Thanks!

(now it takes 40m to finish pre-submit, let's see how long it takes once the action is updated.)